neutrons/SNAPRed
# Ewm6834 fix default record

## Description of work

This pr allows:
- snapred to utilize the default calib record written out as part of state init
- reduction to refer to the correct calibrant sample when creating peaks to purge from normalization

This needs a follow up to address when a state is totally uninitialized.

## Explanation of work


I brought over some changes from `staging` for the calibrant sample stuff.  These were actually somewhat essential when interacting with the default record, as it doesnt have a calibrant sample set (nor is it relevant to reduction).

I updated `GroceryService` and `Indexer` to account for the default record, adding a case to attempt to read on if no other record is available.  Doing this, I also changed the way we pull diffcaltable names, and refer to the calibration record filtering on ws type--this simplified referring to the special name default tables recieve and allow me to avoid special logic in this case.

## To test

Initialize a new state.
run reduction
it should prompt you to continue anyway
see that it completes
```
For reduction/calib
Runnumber: 59039 

for calib:
sample: LA11b6
Grouping: Column
Peak Intensity Thresh: 0.01
Chi: 200
dmin: 1

for norm:
rnunumber: 58810
bgrunnumber: 58813
```

this should apply the default calibration during reduction.

## Link to EWM item


[EWM#6834](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=6834)
